src/main/java/org/kainos/ea/service/JobRoleService.java

```diff
@@ -27,4 +29,23 @@ public List<JobRole> getAllJobRoles() throws FailedToGetJobRolesException {
         }
     }
 
+    public JobRoleSpec getSpecificationById(int id) throws FailedToGetValidJobId {
+        if (id < 0) {
+            throw new FailedToGetValidJobId();
+        }
+
+        try {
+            JobRoleSpec Spec = jobRoleDao.getSpecificationById(id, databaseConnector.getConnection());
+            if (Spec == null) {
+                throw new FailedToGetValidJobId();
+            }
+            return Spec;
```

Comment:
you shouldn't use the same exception for JobRoleDoesNotExist as the SQLException below as these should return 2 different error codes from the API